libs/ast_generic/AST_generic.ml

```diff
@@ -1991,8 +1998,21 @@ let basic_id_info ?(hidden = false) resolved =
 
 (* TODO: move AST_generic_helpers.name_of_id and ids here *)
 
-let dotted_to_canonical xs = Common.map fst xs
-let canonical_to_dotted tid xs = xs |> Common.map (fun s -> (s, tid))
+let canonical_append { unqualified; _ } { unqualified = unqualified2; tok } =
```

Comment:
I think this needs some comment. Why do we append canonical names, and why you can just throw away one of the tokens?

Reply:
There is exactly one point when we need to append canonical names, and that is purely because we are already given an `ImportedEntity` and must combine it with another canonical name. Since we already have the canonical name, we cannot go further upstream to intercept it and avoid appending while they are both canonical.

I think the point of leaving this here, however, is that we have one place we can easily change if we want to change that behavior. Per my other comment, if we want to keep the tokens, we can just change `dotted_to_canonical`. In any event, I don't believe we do need it.

I will add a comment explaining this.